services/learn-card-network/learn-cloud-service/src/routes/user.ts

```diff
@@ -62,10 +63,14 @@ export const userRouter = t.router({
                 const existing = await getUserForDid(did);
 
                 if (existing) {
-                    throw new TRPCError({
-                        code: 'CONFLICT',
-                        message: 'Did is already associated with a user',
-                    });
+                    if (existing.did !== did || existing.associatedDids.length > 0) {
+                        throw new TRPCError({
+                            code: 'CONFLICT',
+                            message: 'Did is already associated with a user',
+                        });
+                    }
+
+                    await deleteUserByDid(did);
                 }
```

Comment:
I'm slightly confused by what's going on here? Is it supposed to be existing.did !== did?

Reply:
So this guy is pretty confusing, so I put a lengthy comment there helping to explain what's going on.

Basically, if you add some stuff to your cache using did:key (i.e. before you sign up for LCN), then switch over to did:web and want to associate those two accounts, that's what this method is for. However, because of the caching stuff, we are now actually making a user for that first did:key, and this would cause this route to error. This added check simply allows deleting that first simple user which was probably made by accident!

Reply:
Awesome! Yes, thanks for adding an explanatory comment 😎